PsfLauncher/UniqueHkey.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <windef.h>
+#include <handleapi.h>
+#include <winreg.h>
+
+class UniqueHKEY
+{
+public:
+    UniqueHKEY()
+    {
+    }
+
+    ~UniqueHKEY()
+    {
+        if (key != INVALID_HANDLE_VALUE)
+        {
+            RegCloseKey((HKEY)key);
+        }
+    }
+
+    PHKEY operator&()
+    {
+        return (PHKEY)&key;
+    }
+
+    void Set(HKEY value)
+    {
+        key = value;
+    }
+

```

Comment:
this too runs the risk of leaking a handle.  std::unique_ptr is a good pattern to follow